File: Habitica/src/main/java/com/habitrpg/android/habitica/models/Avatar.java
package com.habitrpg.android.habitica.models;


import androidx.annotation.Nullable;

import com.habitrpg.android.habitica.models.user.Outfit;
import com.habitrpg.android.habitica.models.user.Stats;

/**
 * Created by phillip on 29.06.17.
 */

public interface Avatar {
    @Nullable
    String getCurrentMount();

    @Nullable
    String getCurrentPet();

    boolean getSleep();

    @Nullable
    Stats getStats();

    @Nullable
    AvatarPreferences getPreferences();

    @Nullable
    Integer getGemCount();

    @Nullable
    Integer getHourglassCount();

    @Nullable
    Outfit getCostume();
    @Nullable
    Outfit getEquipped();

    boolean hasClass();

    boolean isValid();
}

File: Habitica/src/main/java/com/habitrpg/android/habitica/models/Avatar.kt
package com.habitrpg.android.habitica.models

import com.habitrpg.android.habitica.models.user.Outfit
import com.habitrpg.android.habitica.models.user.Stats

/**
 * Created by phillip on 29.06.17.
 */
interface Avatar {
    val currentMount: String?
    val currentPet: String?
    val sleep: Boolean
    val stats: Stats?
    val preferences: AvatarPreferences?
    val flags: AvatarFlags?
    val gemCount: Int
    val hourglassCount: Int
    val costume: Outfit?
    val equipped: Outfit?
    val hasClass: Boolean
    fun isValid(): Boolean
}
